packages/backend/server/src/__tests__/copilot.spec.ts

```diff
@@ -1257,6 +1260,202 @@ test('CitationParser should replace openai style reference chunks', t => {
   t.is(result, expected);
 });
 
+test('TextStreamParser should format different types of chunks correctly', t => {
```

Comment:
Using fixtures to organize test cases is better(more collaborative and reader-friendly):
* Each type of chunk parsing is tested using a test case.
* Another test case to test the parsing of mixed types.